api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -75,6 +78,16 @@ private String convertClusterName(String workspaceId) {
 
   public ResponseEntity<Cluster> createCluster(String workspaceNamespace,
       String workspaceId) {
+
+    // TODO: enforce access level.
+    // This also enforces registered auth domain.
+    WorkspaceAccessLevel accessLevel;
+    try {
+      accessLevel = WorkspacesController.getWorkspaceAccessLevel(workspaceNamespace, workspaceId);
```

Comment:
I would add here:

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/db/dao/WorkspaceService.java#L9

a method like:

void enforceWorkspaceAccessLevel(workspaceNamespace, workspaceId, WorkspaceAccessLevel)

and call it throughout.

(Saves a lot of boilerplate being repeated; allows for use outside of controllers; and we probably shouldn't have controllers depending on each other.)

Reply:
(Also, better not to use static methods too much -- it makes it hard to write unit tests with mocks.)

Reply:
Agreed. I actually just moved it to the service. (Realized after I pushed this up that it broke there, so did that refactor.)